misc-scripts/xref_mapping/XrefParser/RefSeqGPFFParser.pm

```diff
@@ -17,340 +17,330 @@ limitations under the License.
-
-      # get the right source ID based on $type and whether this is predicted (X*) or not
-      my $source_id;
-      if ($type =~ /dna/) {
-	if ($acc =~ /^XM_/ ){
-	  $source_id = $pred_mrna_source_id;
-	} elsif( $acc =~ /^XR/) {
-	  $source_id = $pred_ncrna_source_id;
-	} elsif( $acc =~ /^NM/) {
-	  $source_id = $mrna_source_id;
-	} elsif( $acc =~ /^NR/) {
-	  $source_id = $ncrna_source_id;
-	}
-      } 
-      elsif ($type =~ /peptide/) {
-	if ($acc =~ /^XP_/) {
-	  $source_id = $pred_peptide_source_id;
-	} else {
-	  $source_id = $peptide_source_id;
-	}
-      }
-      print "Warning: can't get source ID for $type $acc\n" if (!$source_id);
-
-      # Description - may be multi-line
-      my ($description) = $entry =~ /DEFINITION\s+([^[]+)/s;
-      print $entry if (length($description) == 0);
-      $description =~ s/\nACCESSION.*//s;
-      $description =~ s/\n//g;
-      $description =~ s/\s+/ /g;
-      $description = substr($description, 0, 255) if (length($description) > 255);
-
-      my ($seq) = $entry =~ /^\s*ORIGIN\s+(.+)/ms; # /s allows . to match newline
-      my @seq_lines = split /\n/, $seq;
-      my $parsed_seq = "";
-      foreach my $x (@seq_lines) {
-        my ($seq_only) = $x =~ /^\s*\d+\s+(.*)$/;
-        next if (!defined $seq_only);
-        $parsed_seq .= $seq_only;
-      }
-      $parsed_seq =~ s#//##g;    # remove trailing end-of-record character
-      $parsed_seq =~ s#\s##g;    # remove whitespace
+  my @coded_by_list = $genbank_rec =~ /\/coded_by=\"(.*?):/xg;
+
+  my $coded_by = pop @coded_by_list;
+
+  $xref->{PAIR} = $coded_by if defined $coded_by;
+
+  my ($refseq_pair) = $genbank_rec =~ /DBSOURCE\s+REFSEQ: accession (\S+)/x;
+  if (!exists $xref->{PAIR}) {
+    $xref->{PAIR} = $refseq_pair if defined $refseq_pair;
+  }
 
-      ( my $acc_no_ver, $ver ) = split( /\./, $ver );
 
-      $xref->{ACCESSION} = $acc;
-      if($acc eq $acc_no_ver){
-         $xref->{VERSION} = $ver;
+  my @gene_ids = $genbank_rec =~ /db_xref=\"GeneID:(.+?)\"/xg;
+  @gene_ids = uniq( @gene_ids );
+
+  # process existing entrez_gene_ids as dependent xrefs
+  GENEID:
+  foreach my $gene_id (@gene_ids) {
+
+    next GENEID unless (defined $self->{entrez}->{$gene_id});
+
+    push @{$xref->{DEPENDENT_XREFS}}, {
+        SOURCE_ID         => $self->source_id_from_name('EntrezGene'),
+        LINKAGE_SOURCE_ID => $acc_source_id,
+        ACCESSION         => $gene_id,
+        LABEL             => $self->{entrez}->{$gene_id}
+    };
+
+    push @{$xref->{DEPENDENT_XREFS}}, {
+        SOURCE_ID         => $self->source_id_from_name('WikiGene'),
+        LINKAGE_SOURCE_ID => $acc_source_id,
+        ACCESSION         => $gene_id,
+        LABEL             => $self->{entrez}->{$gene_id}
+    };
+
+    next GENEID unless (defined $refseq_pair);
+
+    # split the version number
+    my ($pair_acc, $pair_version) = split(/\./x, $refseq_pair);
+
+    # Add xrefs for RefSeq mRNA as well where available
```

Comment:
I trust you have confirmed there is no easy way of having _upload_xref_object_graphs()_ handle these for you?